src/Features/colors/hooks.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback, useEffect } from "react";
+import { getRandomColor } from "./utilities";
+
+export const useColorRandomizer = (setColor: (color: string) => void) => {
```

Comment:
if this is working it's fine, but ideally if you are creating a custom hook, it's a good idea to have states local to the hook. you can return those states from the hook, react can optimize this. 

for example

```jsx
const [color] = useColorRandomizer();
```

and then have state inside the hook

```jsx
const useColorRandomizer() {
	const [color, setColor] = useState();

	// .... rest

	return [color, setColor]
}
```